### ISSUE TITLE
Communicate point estimates for 95% statistically significant results

### ISSUE THREAD

**Issue by aaronyan** [Automattic/abacus#811]
@demetd proposed an idea to show point estimates for metrics that are 95% statistically significant. What do y'all think?

This may help experimenters:

- Gravitate to only reporting statistically significant results
- Simplify the wording and analysis of metrics that have enough certainty

### To do

- [ ] Discuss whether to add point estimates for 95% statistically significant results
- [ ] Design where to put the point estimates
- [ ] Implement the point estimates

**Comment by aaronyan** [comment]
I really liked this idea at first, because simplifying results for experimenters is something I am always fond of. 

But I actually prefer experimenters upgrade and further their understanding of experiment results rather than us simplify it. The **uncertainty of a metric**, even those that are statistically significant, I think should continue to have the center stage when communicating experiment results.

> Gravitate to only reporting statistically significant results

I also would like experimenters to **report on 95% inconclusive results** if they think it plays a part in drawing conclusions. There are instances where the results are likely skewed in one direction but not statistically significant at 95% (maybe at 90% though), and we should help them incorporate these results into their analysis. Adding point estimates for only 95% may detract from very close-to-significance metrics.

**Comment by demetd** [comment]
> I also would like experimenters to report on 95% inconclusive results if they think it plays a part in drawing conclusions.

> Adding point estimates for only 95% may detract from very close-to-significance metrics.

Can you expand on these two points please? Not sure I understand. We also have funnel and observed data from only Treatment being used to draw conclusions but would like to actively work against this as it is wrong and leads to wrong conclusions.

> But I actually prefer experimenters upgrade and further their understanding of experiment results rather than us simplify it.

The two usually go hand-in-hand :) It is easier to understand simpler things.

**Comment by demetd** [comment]
Btw, the idea is to add this to reporting and not remove CIs from reporting. Similar to how ML Sales results get reported: https://github.com/Automattic/ml-sales/issues/106

Or how medical experiments get reported and academic literature reports on CIs - they are usually defined to be evaluated along with a point estimate (especially since we don't necessarily always have symmetric CIs in statistics). 

Some initial hits from simply googling "CI reporting best practices" [#](https://www.ncbi.nlm.nih.gov/pmc/articles/PMC6734580/) [#](https://www.nature.com/articles/s41433-021-01863-w) - Obviously they deal with more foundational issues but the best practices here is actually reporting the mean as well.

I don't think we should report it all the time, I think that would be opening up a can of worms we would not have the resources to contain but for statistically significant results at a 90%, 95%, 99% or 93% (doesn't matter, it is just conventions and how much risk you are willing to take but below 90% is not recommended) level, I don't see any dangeours misuses. 

In fact, with the point estimate there, the education can now focus on the CIs only as we can note when someone leaves the CIs out, we can help them interpret what wide or narrow CIs mean..etc. The CIs become another element.

**Comment by jessie-ross** [comment]
>  >  But I actually prefer experimenters upgrade and further their understanding of experiment results rather than us simplify it.
> 
> The two usually go hand-in-hand :) It is easier to understand simpler things.

I'm aligned with Aaron (and Yanir) here and for me this is one of the most important points - if experimenters have access to point estimates they'd simply use the point estimate and forget about the CI - we used to have observed data and then we very slightly made it less easy to use - it made an impact.

When they don't have the CI, I've seen experimenters reach out to us, and it becomes an important learning opportunity for them to have an "aha" moment about CIs and learn how to use them in communication.

We've (not me but in terms of Automattic) also had this conversation before:
p8AJCL-EJ-p2

And I also recommend you read this discussion, particularly where me and Don come to an agreement on communicating uncertainty to leadership:
p10gg3-aFD-p2

Don has since looped me in on reviewing his "leadership" level statements so it feels like we came to a good outcome - we better understand the pressures involved around responding to stakeholders, and have come up with solutions to address that. If you disagree with anything there it would be great to discuss and we can review our recommendations - that would be timely for my upcoming work around defining experiment conclusions and conclusion reviews.

**Comment by demetd** [comment]
> When they don't have the point estimate, I've seen experimenters reach out to us, and it becomes an important learning opportunity for them to have an "aha" moment about CIs and learn how to use them in communication.

We could automate this and make it scalable. I wouldn't consider educating dozens of individuals on CIs a goal. The goal, for me, is to make it very difficult to interpret results wrong or make the wrong decision and remove some of the back and forth as opposed to encourage it. 

>  if experimenters have access to point estimates they'd simply use the point estimate and forget about the CI 

I think this is the crux of the issue - and where we disagree is that 1. It is okay to rely on the mean when you are not doing comparisons and your results are conclusive, 2. I believe experimenters already don't care about the CI, especially the new audiences we are trying to reach. We are at a much more foundational level of education than _how to report on conclusive results from ExPlat_. 

But I do think having access to point estimate for the posterior mean all the time can be dangerous (they already have access to point estimates: the CI bounds are also point estimates) this is why we should not show point estimates for the mean for inconclusive results. I'd actually go one step further and never show _any_ point estimates for inconclusive results, mean or not -> i.e. would remove even CI bounds' point estimates for these as they tend to get wrongly interpreted, too. But that may be a bit too radical (though I'd def remove point estimates for 50% CIs).

But once the results are conclusive, almost _all_ experimenters will use simply the binary 0/1 (conclusive vs not) to make a decision and will not think through the uncertainty represented by the CI. And that is fine, if we have gotten them that far, that is already a success. 

^ This is the best case scenario, btw, because we have a bigger problem in using the CI to report on inconclusive results because they "tend in the right direction". And an even bigger problem in setting ourselves up for success when designing an experiment and a muuch much bigger problem in understanding when, why, and for what purpose we should run experiments. 

**tl;dr**: I don't think CI-only approach shields us from using experiment results wrong in the slightest, it is not against best practices to report on them, and my hypothesis is that it does partially contribute to ExPlat's reputation as it taking many people, many comments, and too much time to use properly.

I can't access your links but I think I know which convos you are linking to and two remarks:

* They both discuss an all-or-nothing approach: Report all means all the time or never report any of the means. We should not report all posterior means all the time.
* The Don discussion kicked off about reporting observed means - which we should _never_ do. At that point, ExPlat could just become a funnel tool. But I do want to read this discussion in its entirety at some point!

So basically: Not all means are created equal :)

I don't think this is a high priority or urgent issue, so I'll take some time to read more about [best practices in communicating uncertainty](https://www.ncbi.nlm.nih.gov/pmc/articles/PMC6549952/)!

**Comment by aaronyan** [comment]
> Can you expand on these two points please? Not sure I understand. We also have funnel and observed data from only Treatment being used to draw conclusions but would like to actively work against this as it is wrong and leads to wrong conclusions.

Can do! Here's a made-up experiment results example. All credible intervals are 95% and relative lifts for simplicity. The difference in performance between `treatment` and `control` have metrics as follows:

- Performed an important action: +20% to +40% - Treatment winning
- Purchase conversions: -15% to +2% - Inconclusive
- Cash sales: -25% to +10% - Inconclusive

> I also would like experimenters to report on 95% inconclusive results if they think it plays a part in drawing conclusions.

I would want an experiment report to say something like: "`treatment` increased important actions by +20% to +40%, likely decreased purchase conversions although not statistically significant (-15% to +2%), and returned inconclusive cash sales (-25% to +10%).

> Adding point estimates for only 95% may detract from very close-to-significance metrics.

By providing a point estimate for statistically significant results, I think it may make it easier for experimenters to ignore other metrics. In our example, let's add the point estimate to the conclusion part to help it stand out.

- Performed an important action: +20% to +40% - Treatment winning (+30%)
- Purchase conversions: -15% to +2% - Inconclusive
- Cash sales: -25% to +10% - Inconclusive

An experimenter might only look for statistically significant metric results to inform their decision and come up with a report like "`treatment` increased the important action by +30% and the other metrics were inconclusive."

**Comment by aaronyan** [comment]
> I'd actually go one step further and never show any point estimates for inconclusive results, mean or not

Haha we used to hide inconclusive results way back in the early days. I fought against this because I think it feels bad to not be shown the data you collect after the effort of designing, setting up, and implementing an experiment. In a lot of ways, I think the experimenter owns the data they collect, so I think we should show it to them even for inconclusive results.

> tl;dr: I don't think CI-only approach shields us from using experiment results wrong in the slightest, it is not against best practices to report on them, and my hypothesis is that it does partially contribute to ExPlat's reputation as it taking many people, many comments, and too much time to use properly.

I agree that the current way of displaying data is not friendly to most a12s and has likely created a reputation. Even fellow datums like Greg have shown frustration with our designs in the past (hopefully we've made strides since then, pbmo2S-Fn-p2#comment-1581) and Rob has expressed suggestions to simplify top-level results (pbmo2S-UZ-p2#comment-2115). I've also had conversations with Debbie recently on experiment results, and it's clear to me that the current results view is very dense and has a steep learning curve.

I think adding point estimates would be beneficial, but I'm not sure it works well with the current way we tell our story with experiment data (my fears [here](https://github.com/Automattic/abacus/issues/811#issuecomment-1334700427)). I do think stat sig point estimates have a place if we were to redesign the way we show results and tell a story with data that gets more complicated the more you drill down (pbmo2S-UZ-p2 ideas like observed data first). Because we currently show top-level results as complicated first, I fear that folks will gravitate towards the point estimates too fast and forget about inconclusive metrics.

**Comment by jessie-ross** [comment]
> I don't think this is a high priority or urgent issue

Yeah let's leave this be for the moment, I'll just note that I have a lot more to say on the matter.

**Comment by razvanpapadopol** [comment]
As Jessie [mentioned](https://github.com/Automattic/abacus/issues/811#issuecomment-1333160815), we've made important progress in communicating uncertainty, and if that continues to go in a good direction, I'm not sure if the point estimates will provide much more value.

> Simplify the wording and analysis of metrics that have enough certainty

In order for any related sentence to be correct, we'll need to mention this is based on the observed data and also to add some disclaimer similar to the one used on Estimated Impact.

The [example](https://github.com/Automattic/abacus/issues/811#issuecomment-1334700427) shared above shows where results can be clearly misinterpreted and I remember also real examples (using the old A/B testing tool - Tracks Hypotheses) where decisions were taken by ignoring secondary metrics for which only the attribution window was different (7D was ignored for a winning 1H).

**Comment by demetd** [comment]
> In order for any related sentence to be correct, we'll need to mention this is based on the observed data and also to add some disclaimer similar to the one used on Estimated Impact.

This would not be based on observed data. 

> The https://github.com/Automattic/abacus/issues/811#issuecomment-1334700427 shared above shows where results can be clearly misinterpreted

This is for results that did were not statistically significant. Which can be misinterpreted and should not have point estimates - in fact, I wouldn't even mind not showing anything there as I've mentioned a couple times.

> where decisions were taken by ignoring secondary metrics for which only the attribution window was different (7D was ignored for a winning 1H).

Again, this is not about point estimates of the mean of the posteriors for statistically significant results.

Experiment results have been misinterpreted many times. But not due to using the point estimate for the mean of the posteriors for statistically significant results - since there is a purpose to it. 

Point estimates have been misinterpreted a lot of times. E.g. we had an initial ExPlat post that initially used the point estimates for the lower bound of the 50% CI as a "conservative estimate". So even experts in the area can misinterpret _the point estimates we already show._

So, the suggestion is the following: **Show the point estimate for the mean of the posteriors for statistically significant results** and nothing else.

So far the only argument against this (that is relevant) has been: We like describing uncertainty when people notice that they don't have a point estimate and we think this is useful as it forces people to learn more about uncertainty by reaching out to us.

I think this can be a valid argument so maybe we focus on this as I genuinely don't see how the rest of the arguments are related.

**Comment by razvanpapadopol** [comment]
Thanks for the clarifications, @demetd ! 🙌🏼 

I guess it's anyway better for us to show them correctly and only when they should be taken into consideration instead of experimenters trying to calculate by themselves. 

As long as we don't downgrade (visually, in the UI) the metrics with inconclusive results, we continue to display the CIs, and we review the experiment conclusions, we should be on the safe side.

> Design where to put the point estimates

Does it make sense to put them where we say the results for some metric are actionable? With time, if we notice positive feedback and usage, we can move the value to the top-level results.

<img width="1236" alt="Screenshot 2022-12-08 at 19 10 45" src="https://user-images.githubusercontent.com/14192054/206520552-a0250959-9859-4ac7-80cb-1749a623cab1.png">
